### DISCUSSION TITLE
Question about the allowed signature for a forecaster's predict_interval() method. (Motivated by wanting to do an enhancement.)

### DISCUSSION THREAD

**Discussion by ericjb** [alan-turing-institute/sktime#6268] [Q&A]
I am interested in enhancing the PolynomialTrendForecaster forecaster by implementing the predict_interval() method for it. My implementation requires as an input the index of the time series used to fit the model. Is it possible to add this as an additional argument to predict_interval()?

**Comment by fkiraly** [comment]
Interesting thought.

Some questions and responses below.

> Is it possible to add this as an additional argument to `predict_interval`?

It's not necessary, because there is a simple way you can get access to it:
you can have the estimtor memorize whatever information you need, in `_fit`. Simply write it to an attr ending in underscore.

(design-wise, it would also not be desirable, since the user would have to "feed" information that the estimator should already know)

> enhancing the `PolynomialTrendForecaster` forecaster by implementing the `predict_interval` method for it.

May I ask what your idea is, here? There might not be a "canonical" way to do this, but of course adding this - perhaps as a new estimator - is welcome.

**Comment by ericjb** [comment]
In the context of time series, PolynomialTrendForecaster is simply an OLS where we add the explanatory variables t, t^2, ..., t^p and possibly t^0 if we allow an intercept, where p is the degree of the polynomial. If we make the standard assumption that the residuals of the fit are IID N(0, sigma^2), we use the fit to estimate sigma^2. A forecast has a Student's t-distribution with n-p (or n - (p+1)) d.f. where n = # of observations in the training set. It's fairly straightforward to carry this out. I wasn't sure that _fit was allowed to retain information, but if that's allowable then all is good. Thanks for the quick response.

**Reply by fkiraly** [reply]
hm, is this not sth that would be applicable to all forecasters, rather than just the polynomial trend one? As long as the assumption on iid residuals is satisfied.

Similar to `NaiveVariance` in `forecasting.naive` which can take any forecaster internally?

**Reply by fkiraly** [reply]
oh, I see, only linear ones, probably.

> I wasn't sure that _fit was allowed to retain information, but if that's allowable then all is good.

Of course, how else would you "fit" 😄
